core/object/object.h

```diff
@@ -801,6 +801,8 @@ class Object {
 		NOTIFICATION_POSTINITIALIZE = 0,
 		NOTIFICATION_PREDELETE = 1,
 		NOTIFICATION_EXTENSION_RELOADED = 2,
+		// Internal notification to send after NOTIFICATION_PREDELETE, not bound to scripting.
+		NOTIFICATION_PREDELETE_CLEANUP = 3,
```

Comment:
This needs to be clearly reserved so future notifications don't use it, especially if we use a system based on registered values to track overlap

Reply:
It seems we use different _magnitudes_[^1] to avoid overlap (e.g.: the Node notifications are 1X, CanvasItem notifications are 3X, Control/Node3D notifications are 4X, Window notifications are 10XX). But I agree it can be error prone.

[^1]: Not sure if _magnitude_ is the right word.